src/main/webapp/cdn/blockly/generators/propc/communicate.js

```diff
@@ -1486,4 +1486,170 @@ Blockly.propc.ws2812b_update = function() {
     } else {
         return 'ws2812_set(__ws2812b, LED_PIN, RGBleds, LED_COUNT);\n';
     }
-};
\ No newline at end of file
+};
+
+// --------------------- Simple WX Module --------------------------------------
+Blockly.Blocks.wx_set_widget = {
+    init: function() {
+        this.setColour(colorPalette.getColor('protocols'));
+        this.appendDummyInput("SET_1")
+            .appendField("Simple WX set widget")
+            .appendField(new Blockly.FieldDropdown([["1", "1"], ["2", "2"], ["3", "3"], ["4", "4"], ["5", "5"], ["6", "6"], ["7", "7"], ["8", "8"], ["9", "9"], ["10", "10"], ["11", "11"], ["12", "12"]]), "WIDGET")    
+            .appendField("to a")
+            .appendField(new Blockly.FieldDropdown([
+                ["Button \u2794", '0'], 
+                ["Switch \u2794", '1'], 
+                ["Slider \u2794", '2'], 
+                ["Send Value \u2794", '3'], 
+                ["Pick Color \u2794", '4'], 
+                ["\u2794 Show Value", '5'], 
+                ["\u2794 Gauge", '6'], 
+                ["\u2794 Bar Graph", '7'], 
+                ["\u2794 Show Color", '8'], 
+                ["\u2794 Light Bulb", '9'], 
+                ["Clear Widget", '10']], function (type) {
+                    this.sourceBlock_.updateShape_({"TYPE": type});
+                }), "TYPE");
+        this.appendDummyInput("SET_2")
+            .appendField("widget color")
+            .appendField(new Blockly.FieldColour("#ffffff"), "COLOR")
+            .appendField(" minimum")
+            .appendField(new Blockly.FieldTextInput('0', Blockly.FieldTextInput.numberValidator), 'MIN')
+            .appendField(" maximum")
+            .appendField(new Blockly.FieldTextInput('10', Blockly.FieldTextInput.numberValidator), 'MAX')
+            .appendField(" initial value")
+            .appendField(new Blockly.FieldTextInput('5', Blockly.FieldTextInput.numberValidator), 'INITIAL');
+        this.setInputsInline(false);
+        this.setPreviousStatement(true, null);
+        this.setNextStatement(true, null);
+    },
+    mutationToDom: function () {
+        var container = document.createElement('mutation');
+        var type = this.getFieldValue('TYPE');
+        container.setAttribute('TYPE', type);
+        return container;
+    },
+    domToMutation: function (xmlElement) {
+        var type = xmlElement.getAttribute('TYPE');
+        this.updateShape_({"TYPE": type});
+    },
+    updateShape_: function (details) {
+        var type = details['TYPE'];
+        if (details['TYPE'] === undefined) {
+            type = this.getFieldValue('TYPE');
+        }
+
+        if (this.getInput('SET_2') !== undefined) {
+            this.removeInput('SET_2');
+        }
+        var inputPins;
+        if (type !== '10') {
+            this.appendDummyInput("SET_2");
+            inputPins = this.getInput('SET_2');
+        }
+        if (type === '2' || type === '6' || type === '7') {
```

Comment:
Can this be converted to a switch statement with constant_names instead of type values?